Binding for octoller (www.octoller.com) V1.0.0 / Try 2

JPlenert opened the pull request:
New binding for octoller. octoller is an open source, smart home and programmable logic controller(www.octoller.com). Right now the binding needs an .Net gateway (octoller.gateway). In a future version it may communicate directly to the octoller.

buildhive commented:
[openhab » openhab #2918](https://buildhive.cloudbees.com/job/openhab/job/openhab/2918/) FAILURE
Looks like there's a problem with this pull request
[(what's this?)](https://www.cloudbees.com/what-is-buildhive)

teichsta commented on bundles/binding/org.openhab.binding.octoller/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,30 @@
+Bundle-Name: openHAB octoller Binding
+Bundle-SymbolicName: org.openhab.binding.octoller
+Bundle-Vendor: openHAB.org
+Bundle-Version: 1.0.0
```
1.8.0

teichsta commented on bundles/binding/org.openhab.binding.octoller/OSGI-INF/binding.xml:
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2010-2014, openHAB.org and others.
```
-2015

teichsta commented on bundles/binding/org.openhab.binding.octoller/OSGI-INF/genericbindingprovider.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Copyright (c) 2010-2014, openHAB.org and others.
```
-2015

teichsta commented on bundles/binding/org.openhab.binding.octoller/pom.xml:
```diff
@@ -0,0 +1,42 @@
+	<parent>
+		<groupId>org.openhab.bundles</groupId>
+		<artifactId>binding</artifactId>
+		<version>1.5.1-SNAPSHOT</version>
```
1.8.0-SNAPSHOT

teichsta commented on bundles/binding/org.openhab.binding.octoller/pom.xml:
```diff
@@ -0,0 +1,42 @@
+		</plugins>
+	</build>
+
+	<url>www.octoller.com</url>
```
remove

teichsta commented on bundles/binding/org.openhab.binding.octoller/pom.xml:
```diff
@@ -0,0 +1,42 @@
+For use with octoller-Gateway</description>
+	<organization>
+		<name>Joerg Plenert</name>
+	</organization>
```
remove

teichsta commented on bundles/binding/org.openhab.binding.octoller/pom.xml:
```diff
@@ -0,0 +1,42 @@
+	<organization>
+		<name>Joerg Plenert</name>
+	</organization>
+	<version>1.0</version>
```
remove

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/octoller/devicecom/Connection.java:
```diff
@@ -0,0 +1,115 @@
+ * openHAB binding for octoller (www.octoller.com)
+ * For use with octoller-Gateway
+ ********************************************************************************************************
+ */
```
remove this comment block

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/octoller/devicecom/Connection.java:
```diff
@@ -0,0 +1,115 @@
+
+/**
+ * @author JPlenert
+ * @since 1.5.1
```
1.8.0

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/octoller/devicecom/Connection.java:
```diff
@@ -0,0 +1,115 @@
+ * @author JPlenert
+ * @since 1.5.1
+ * openHAB binding for octoller (www.octoller.com)
+ * For use with octoller-Gateway
```
move comment above @author/@version tag according to the java coding conventions (see https://github.com/openhab/openhab/wiki/How-To-Contribute)

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/octoller/devicecom/Connection.java:
```diff
@@ -0,0 +1,115 @@
+		return sb.toString();
+	}
+	
+}
```
add newline

teichsta commented:
Hi @JPlenert,

sorry for letting you waiting soooo long before starting the review. While going through the code i am wondering whether you've read the [How To Contribute](https://github.com/openhab/openhab/wiki/How-To-Contribute) guide. While continuing the code review could you please:

* apply the code formatter (right click > source > format) to all of your files
* update version to 1.8.0
* update the license header
* add a description to all of the classed and remove additional comments below @author/@since

Thanks in advance,

Thomas E.-E.

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/openhab/binding/octoller/internal/octollerBinding.java:
```diff
@@ -0,0 +1,174 @@
+					logger.debug(itemName + " returned " + result);
+				}
+				catch (Exception e) {
+					e.printStackTrace();
```
replace with proper logging

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/openhab/binding/octoller/internal/octollerBinding.java:
```diff
@@ -0,0 +1,174 @@
+				if (config == null)
+					continue;
+					
+				//eventPublisher.postUpdate(itemName, (State)new PercentType(100));
```
remove commented code

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/openhab/binding/octoller/internal/octollerBinding.java:
```diff
@@ -0,0 +1,174 @@
+		// event bus goes here. This method is only called if one of the 
+		// BindingProviders provide a binding for the given 'itemName'.
+		logger.debug("internalReceiveCommand() is called!");
+	}
```
remove overridden method

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/openhab/binding/octoller/internal/octollerBinding.java:
```diff
@@ -0,0 +1,174 @@
+				refreshInterval = Long.parseLong(refreshIntervalString);
+			}
+			
+			// read further config parameters here ...
```
remove generic comment

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/openhab/binding/octoller/internal/octollerBindingConfig.java:
```diff
@@ -0,0 +1,24 @@
+
+/**
+ * @author JPlenert
+ * @since 1.5.1
```
1.8.0

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/openhab/binding/octoller/internal/octollerBindingConfig.java:
```diff
@@ -0,0 +1,24 @@
+
+import org.openhab.core.binding.BindingConfig;
+
+/**
```
briefly explain the purpose of this class

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/openhab/binding/octoller/internal/octollerBindingConfig.java:
```diff
@@ -0,0 +1,24 @@
+	public String DeviceHost;
+	public String GatewayHost;
+	public String BlockName;
+	public int BlockID;
```
explain what the members are about

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/openhab/binding/octoller/internal/octollerGenericBindingProvider.java:
```diff
@@ -0,0 +1,92 @@
+
+/**
+ * This class is responsible for parsing the binding configuration.
+ * For openHAB binding for octoller (www.octoller.com)
```
remove generic comment and briefly explain what this class is about, also add some Binding configuration examples here.

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/openhab/binding/octoller/internal/octollerGenericBindingProvider.java:
```diff
@@ -0,0 +1,92 @@
+ * For openHAB binding for octoller (www.octoller.com)
+ *  
+ * @author JPlenert
+ * @since 1.5.1
```
1.8.0

teichsta commented on bundles/binding/org.openhab.binding.octoller/src/main/java/org/openhab/binding/octoller/octollerBindingProvider.java:
```diff
@@ -0,0 +1,22 @@
+
+/**
+ * @author JPlenert
+ * @since 1.5.1
```
1.8.0

teichsta commented:
i am done with the review. Please find most of the comments inline in addition to the general comments above.

Due to our [1.8 schedule](https://community.openhab.org/t/upcoming-1-8-release/4892/1) i would kindly like to ask you if you would be able to incorporate my feedback until December 30th? Sorry for this hustle as i was the one letting you waiting so long :-).

Thanks in advance, Thomas E.-E.

JPlenert commented:
Hi Thomas,
I just made the changes. Unfortunately, I’m not able to compile any newer version of openhab. 
After redoing the whole process with maven there are still about 1000 compile errors and missing files even without compiling. Yes, I also changed the Compiler compliance level to 1.7 (and 1.6) and changed the JRE. Maybe this is a problem on Windows 10? With 1.5 on Windows 7 I had no problems.
Hopefully that will change with the final release of 1.8…
However; I had no chance to do any tests!

Regards Joerg

P.S. Have I ever said before said I love C/C++/C#? :-)

Some more details on the problem:

Maven process was OK:

[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 29:06 min
[INFO] Finished at: 2015-12-28T12:06:00+01:00
[INFO] Final Memory: 435M/918M
[INFO] ------------------------------------------------------------------------

Errors in Eclipse without compile:
Description	Resource	Path	Location	Type
Project 'org.openhab.binding.urtsi' is missing required source folder: 'src/main/xtend-gen'	org.openhab.binding.urtsi		Build path	Build Path Problem
Project 'org.openhab.model.persistence.ui' is missing required source folder: 'src-gen'	org.openhab.model.persistence.ui		Build path	Build Path Problem
Project 'org.openhab.persistence.jdbc' is missing required library: 'lib/commons-dbutils-1.6.jar'	org.openhab.persistence.jdbc		Build path	Build Path Problem
Project 'org.openhab.persistence.jdbc' is missing required library: 'lib/HikariCP-2.4.2.jar'	org.openhab.persistence.jdbc		Build path	Build Path Problem
Project 'org.openhab.persistence.jdbc' is missing required library: 'lib/yank-3.1.0.jar'	org.openhab.persistence.jdbc		Build path	Build Path Problem
Project 'org.openhab.persistence.jpa' is missing required library: 'lib/derbyclient.jar'	org.openhab.persistence.jpa		Build path	Build Path Problem
Project 'org.openhab.persistence.jpa' is missing required library: 'lib/mysql-connector-java-5.1.26-bin.jar'	org.openhab.persistence.jpa		Build path	Build Path Problem
Project 'org.openhab.persistence.jpa' is missing required library: 'lib/openjpa-all-2.3.0.jar'	org.openhab.persistence.jpa		Build path	Build Path Problem
Project 'org.openhab.persistence.jpa' is missing required library: 'lib/postgresql-9.2-1004.jdbc41.jar'	org.openhab.persistence.jpa		Build path	Build Path Problem

JPlenert commented:
I just tried to use the vagrant IDE to give it a try. That doesn't work also, because the SSL certificate of "atlas.hashicorp.com" is no longer valid. My IE & Edge also report a broken certificate. No luck!

==> default: Adding box 'rub-a-dub-dub/openhabdev32' (v0.1.0) for provider: virtualbox
    default: Downloading: https://atlas.hashicorp.com/rub-a-dub-dub/boxes/openhabdev32/versions/0.1.0/providers/virtualb
ox.box
    default: Progress: 100% (Rate: 176/s, Estimated time remaining: --:--:--)
An error occurred while downloading the remote file. The error
message, if any, is reproduced below. Please fix this error and try
again.

SSL certificate problem: certificate has expired

teichsta commented:
Thanks @JPlenert!
